### ISSUE TITLE
🐛 The `gradient_accumulation_steps` in the training parameters doesn't work.

### ISSUE THREAD

**Issue by yubinCloud** [starVLA/starVLA#86]
When training model and setting the parameter `gradient_accumulation_steps` in the startup script, this parameter **did not take effect**. The startup command is as follows:

```bash
accelerate launch \
    ...
    --trainer.gradient_accumulation_steps 2 \
    ...
```

Then, when we print `self.accelerator.gradient_state` during training, we can see that `num_steps=1` for gradient accumulation:

<img width="351" height="67" alt="Image" src="https://github.com/user-attachments/assets/ed74da3e-d6f5-4cf3-aeaa-ba50e82467d4" />

**This bug arises because the relevant parameters for gradient accumulation were not passed in during the initialization of accelerator, while the training process relies on the accelerator.** (ref: [Accelerate Document](https://huggingface.co/docs/accelerate/usage_guides/gradient_accumulation))

**Comment by JinhuiYE** [comment]
Yes, gradient_accumulation_step can be set in starVLA/config/starVLA/config/deepseeds/ds_config.yaml

```yaml
    "train_micro_batch_size_per_gpu": "auto",
    "train_batch_size": "auto",
    "gradient_accumulation_steps": 1,
```

**Comment by Tgzz666** [comment]
> 是的，gradient_accumulation_step可以在 starVLA/config/starVLA/config/deepseeds/ds_config.yaml 中设置
> 
>     "train_micro_batch_size_per_gpu": "auto",
>     "train_batch_size": "auto",
>     "gradient_accumulation_steps": 1,

作者您好，我发现deepspeed config中设置 gradient accumulate 仍然不起作用，从epooch计算进度中发现实际batch中gradient accumulate恒为1

**Comment by JinhuiYE** [comment]
这个设置会基于 deepseeds 底层实现同步的， print 的消息是看不到的。

世界上这样会导致 real gradient_accumulation_steps = deepseeds 内部的 * 直接在 trainer 中实现的的

**Comment by LanesraL** [comment]
> Yes, gradient_accumulation_step can be set in starVLA/config/starVLA/config/deepseeds/ds_config.yaml
> 
>     "train_micro_batch_size_per_gpu": "auto",
>     "train_batch_size": "auto",
>     "gradient_accumulation_steps": 1,

@JinhuiYE Hi! While reviewing the training code, I noticed that changing trainer.gradient_accumulation_steps in the training config does not seem to have any effect during training. The value only takes effect when it is modified in the DeepSpeed (ds) configuration.

Could keeping gradient_accumulation_steps in the training config pose a potential risk of misleading users? If my understanding or inspection is incorrect, I would appreciate your clarification.
